bleach/sanitizer.py

```diff
@@ -596,44 +597,34 @@ def disallowed_token(self, token):
 
     def sanitize_css(self, style):
         """Sanitizes css in style tags"""
-        # Convert entities in the style so that it can be parsed as CSS
-        style = html5lib_shim.convert_entities(style)
-
-        # Drop any url values before we do anything else
-        style = re.compile(r"url\s*\(\s*[^\s)]+?\s*\)\s*").sub(" ", style)
-
-        # The gauntlet of sanitization
-
-        # Validate the css in the style tag and if it's not valid, then drop
-        # the whole thing.
-        parts = style.split(";")
-        gauntlet = re.compile(
-            r"""^(  # consider a style attribute value as composed of:
-[/:,#%!.\s\w]    # a non-newline character
-|\w-\w           # 3 characters in the form \w-\w
-|'[\s\w]+'\s*    # a single quoted string of [\s\w]+ with trailing space
-|"[\s\w]+"       # a double quoted string of [\s\w]+
-|\([\d,%\.\s]+\) # a parenthesized string of one or more digits, commas, periods, ...
-)*$""",  # ... percent signs, or whitespace e.g. from 'color: hsl(30,100%,50%)'
-            flags=re.U | re.VERBOSE,
-        )
-
-        for part in parts:
-            if not gauntlet.match(part):
-                return ""
+        parsed = tinycss2.parse_declaration_list(style)
```

Comment:
https://doc.courtbouillon.org/tinycss2/stable/api_reference.html#tinycss2.parse_stylesheet_bytes

I wasn't sure whether to enable `trim_whitespace` and `trim_comments`. The output of `trim_whitespace` wipes all the whitespace tokens and it doesn't look good. However, when we remove tokens because they're not in the allow list, we end up with a series of consecutive whitespace tokens. I accounted for that by looking at the last token in the `new_tokens` list (poorly named) to see if it's a whitespace token and ignoring the new one if it is. Maybe that's good enough?